src/onegov/search/mixins.py

```diff
@@ -161,52 +163,46 @@ def search_score(self):
         return 10
 
     @staticmethod
-    def psql_tsvector_string(model):
+    def psql_tsvector_expression(model):
         """
-        Provides the tsvector string for postgres defining which columns and
-        json fields to be used for full text search index.
+        Provides the tsvector expression for postgres for the defined
+        model. Depending on the model columns and properties are used for full
+        text search index.
 
-        Example:
-        s = create_tsvector_string('title', 'location')
-        s += " || ' ' || coalesce(((content ->> 'description')), '')"
-        s += " || ' ' || coalesce(((content ->> 'organizer')), '')"
-
-        :return: tsvector string
+        :return: tsvector expression
         """
-
-        columns = [p for p in model.es_properties if p in model.__dict__
-                   and not p.startswith('es_')]
-        return Searchable.create_tsvector_string(*columns) if columns else ''
+        columns = [p for p in model.es_properties if
+                   p in model.__dict__ and not p.startswith('es_')]
+        objects = [getattr(model, col) for col in columns]
```

Comment:
```suggestion
        objects = [getattr(model, p) for p in model.es_properties]
```
if `es_properties` contains attributes that don't exist on the model that should be an error.

Also as a side note, you are currently ignoring the type of the `es_properties`, even though some of them can be represented in tsvector expressions (e.g. `translated_text` vs `text`). There may also be a way to treat `keywords` correctly. On that note `keywords` might be a JSON array, so you would need to special case it anyways.

Reply:
if `es_property` from type `keyword` a property it works well, see directory_entry

Reply:
done